src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/LockFileUtils.cs

```diff
@@ -330,34 +330,53 @@ public static class LockFileUtils
                 // Add files under asset groups
                 object filesObject;
                 object msbuildPath;
-                if (localMatch.LocalLibrary.Items.TryGetValue(KnownLibraryProperties.ProjectRestoreMetadataFiles, out filesObject)
-                     && localMatch.LocalLibrary.Items.TryGetValue(KnownLibraryProperties.MSBuildProjectPath, out msbuildPath))
+                if (localMatch.LocalLibrary.Items.TryGetValue(KnownLibraryProperties.MSBuildProjectPath, out msbuildPath))
                 {
-                    var files = (List<ProjectRestoreMetadataFile>)filesObject;
+                    var files = new List<ProjectRestoreMetadataFile>();
                     var fileLookup = new Dictionary<string, ProjectRestoreMetadataFile>(StringComparer.OrdinalIgnoreCase);
 
-                    foreach (var file in files)
+                    // Find the project path, this is provided by the resolver
+                    var msbuildFilePathInfo = new FileInfo((string)msbuildPath);
+
+                    // Ensure a trailing slash for the relative path helper.
+                    var projectDir = msbuildFilePathInfo.Directory.FullName
```

Comment:
we have a helper method in PathUtility for this..perhaps use that instead for consistency?